File: Leetcode - Java/1299_array_replaceElements.java
// https://leetcode.com/problems/replace-elements-with-greatest-element-on-right-side/

class Solution {
    public int[] replaceElements(int[] arr) {
        int size = arr.length-1;
        int prevElem = arr[size];
        arr[size] = -1; 
        for(int i = size - 1; i >= 0; i--) {
            int currElem = Math.max(prevElem, arr[i+1]);
            prevElem = arr[i];
            arr[i] = currElem;
        }
        return arr;
    }
}

File: Leetcode - Python/1299_array_replaceElements.py
# https://leetcode.com/problems/replace-elements-with-greatest-element-on-right-side/


class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        size = len(arr) - 1
        prevElem = arr[size]
        arr[size] = -1

        for i in range(size - 1, -1, -1):
            currElem = max(prevElem, arr[i + 1])
            prevElem = arr[i]
            arr[i] = currElem

        return arr
